### PR TITLE
Springboot/trader account controller

### PR DESCRIPTION
**Pull request by AkshaySapra**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 intellij-java-google-style.xml
 **/target/
 **/.idea
+api-docs.json
+trading.iml
```
Comment (jarviscanada) [inline .gitignore:5]: https://www.notion.so/jarviscanada/Git-Github-GitFlow-Guidelines-4254192fe01e41bbbb2cce3f85185cda#4ea9e5dd98844e1286332e5c4eb5cec4

see this tip

File: springboot/src/main/java/ca/jrvs/apps/trading/dao/JdbcCrudDao.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        return out;
+    }
+    private <S extends T> void addOne(S entity){
+        SqlParameterSource parameterSource = new BeanPropertySqlParameterSource(entity);
+        Object newId = getSimpleJdbcInsert().executeAndReturnKey(parameterSource);
+        entity.setId((Integer) newId);
+    }
+
+
+    abstract public int updateOne(T entity);{
```
Comment (jarviscanada) [inline springboot/src/main/java/ca/jrvs/apps/trading/dao/JdbcCrudDao.java]: - fix your coding style. you don't need {}. `abstract public int updateOne(T entity);`
- remove extra newlines (e.g. line 58). it looks weird when you have inconsistent newlines
- autoformat source code files in batch https://www.notion.so/jarviscanada/IntelliJ-IDEA-Guide-2e124ee013ee4ef3b8e1ee55e3b494aa#ec04f4fa69974f5facf996fae8a85d4d

File: springboot/src/main/java/ca/jrvs/apps/trading/dao/JdbcCrudDao.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    public List<T> findAllById(Iterable<Integer> ids){
+        List<T> entities = new ArrayList<>();
+        for (Integer id : ids){
+            entities.add(findById(id).get());
+        }
+        return entities;
+    }
+
+
+
```
Comment (jarviscanada) [inline springboot/src/main/java/ca/jrvs/apps/trading/dao/JdbcCrudDao.java]: again. remove all extra newlines

File: springboot/src/main/java/ca/jrvs/apps/trading/service/TraderAccountService.java [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+                                SecurityOrderDao securityOrderDao, PositionDao positionDao){
+        this.traderDao = traderDao;
+        this.accountDao = accountDao;
+        this.securityOrderDao = securityOrderDao;
+        this.positionDao = positionDao;
+    }
+
+
+    public TraderAccountView createTraderAndAccount(Trader trader) throws IllegalArgumentException{
+        if (trader == null){ throw new IllegalArgumentException("The trader cannot be null.");}
```
Comment (jarviscanada) [inline springboot/src/main/java/ca/jrvs/apps/trading/service/TraderAccountService.java]: use your IDE to autofromat

Comment (jarviscanada) [reply springboot/src/main/java/ca/jrvs/apps/trading/service/TraderAccountService.java]: use OR 
e.g. `if (trader.getCountry() == null || trader.getDob() == null `

File: springboot/src/main/java/ca/jrvs/apps/trading/service/TraderAccountService.java [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        return positions;
+    }
+
+    private boolean hasOpenPositions(List<Position> positionList) {
+        for (Position position : positionList) {
+            if (position.getPosition() != 0) {
+                return true;
+            }
+        }
+        return false;
```
Comment (jarviscanada) [inline springboot/src/main/java/ca/jrvs/apps/trading/service/TraderAccountService.java]: try. to use STREAM API

File: springboot/src/test/java/ca/jrvs/apps/trading/dao/AccountDaoIntTest.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                )
+        );
+        assertEquals(1, accounts.size());
+        assertEquals(account.getAmount(), accounts.get(0).getAmount());
+        assertEquals(account.getTrader_id(), accounts.get(0).getTrader_id());
+        assertEquals(account.getId(),accounts.get(0).getId());
+
+    }
+
+
```
Comment (jarviscanada) [inline springboot/src/test/java/ca/jrvs/apps/trading/dao/AccountDaoIntTest.java]: clean up newlines

**Review by jarviscanada — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (AkshaySapra) [reply .gitignore:5]: @jarviscanada Yes, I added it to my gitignore but it appears its still version controlled even after deleting the files and adding them again. I am going to do [this](https://gist.github.com/tsrivishnu/a2f3adbbca9fcad5f3597af301ad1abb)

File: springboot/src/main/java/ca/jrvs/apps/trading/service/TraderAccountService.java (hunk 4 above)
Comment (AkshaySapra) [reply springboot/src/main/java/ca/jrvs/apps/trading/service/TraderAccountService.java]: @jarviscanada But then I cannot give a specific exception to give info on which parameter is not valid?

**Pull request merged**